src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraFlight2B2T.kt

```diff
@@ -0,0 +1,346 @@
+                    true))
+            } else {
+                /* Position needs to be sent to the server */
+                player.connection.sendPacket(CPacketPlayer.Position(player.posX,
+                    player.posY,
+                    player.posZ,
+                    true))
+            }
+        } else {
+            /* Position and rotation need to be sent to the server */
+            player.connection.sendPacket(CPacketPlayer.PositionRotation(player.posX,
+                player.posY,
+                player.posZ,
+                player.pitchYaw.y,
+                player.pitchYaw.x,
+                true))
+        }
+
+        lastRotation = rotation
+        lastPos = player.positionVector
+    }
+
+    /**
+     * Calculate the starting height. Constantly update the position as we would in vanilla until the correct criteria
+     * is met. Should only be called from the state NOT_STARTED.
+     */
+    private fun SafeClientEvent.calcStartingHeight() {
+        /* We are in the air at least 0.5 above the ground and have an elytra equipped */
+        if (!player.onGround && player.inventory.armorInventory[2].item == Items.ELYTRA && (player.posY - getGroundPos().y >= jumpHeightMin)) {
+            /* Start at pos X.5 or higher */
+            if (isJumpStart && (player.posY - player.posY.toInt() >= jumpHeightMin)) {
+                /* Jumping start */
+                originIdle = player.positionVector
+                started = MovementState.IDLE
+                idleStart = System.currentTimeMillis()
+                if (showDebug.value) sendChatMessage("Jump start at height: " + player.posY)
+                player.capabilities.allowFlying = true
+                player.capabilities.isFlying = true
+            } else if (!isJumpStart) {
+                /* Falling start */
+                originIdle = player.positionVector
+                started = MovementState.IDLE
+                idleStart = System.currentTimeMillis()
+                if (showDebug.value) sendChatMessage("Fall start at height: " + player.posY)
+                player.capabilities.allowFlying = true
+                player.capabilities.isFlying = true
+            }
+        }
+    }
+
+    /**
+     * Calculate the idle position. Some movement is needed so that we do not get kicked for flying. Move in a circle.
+     */
+    private fun getNextIdle(): Vec3d {
+        return Vec3d(
+            cos((((System.currentTimeMillis() - idleStart) / idleSpeed.value) % 360).toDouble()) * idleRadius.value,
+            0.0,
+            sin((((System.currentTimeMillis() - idleStart) / idleSpeed.value) % 360).toDouble()) * idleRadius.value,
+        )
+    }
+
+    override fun isActive(): Boolean {
+        return started != MovementState.NOT_STARTED
+    }
+
+    private fun SafeClientEvent.reset() {
+        if (!player.isCreative) player.capabilities.allowFlying = false
+        player.capabilities.isFlying = false
+        player.onGround = true
```

Comment:
Only run the needed codes in SafeClientEvent with `runSafe` so it would reset stuff correctly when player is not in game.
(eg. disabled on disconnected)